maker/maker.go

```diff
@@ -114,32 +165,58 @@ func ParseStruct(src []byte, structName string, copyDocs bool) (methods []Method
 	}
 
 	for _, d := range a.Decls {
-		if a, fd := GetReceiverTypeName(src, d); a == structName {
-			methodName := fd.Name.String()
+
+		if name := GetStructName(src, d); name != "" {
+			if parseds[name] == nil {
+				parseds[name] = &Parsed{}
+			}
+			parseds[name].Embedded = append(parseds[name].Embedded, GetEmbedded(src, d)...)
+			continue
+		}
+
+		a, fd := GetReceiverTypeName(src, d)
+		var methodName string
+		if fd != nil {
+
+			methodName = fd.Name.String()
 			if methodName[0] > 'Z' {
 				continue
 			}
-			params, _ := GetParameters(src, fd.Type.Params)
-			ret, merged := GetParameters(src, fd.Type.Results)
-
-			var retValues string
-			if merged {
-				retValues = fmt.Sprintf("(%s)", strings.Join(ret, ", "))
-			} else {
-				retValues = strings.Join(ret, ", ")
+			if StringSlice(exs).Contain(methodName) {
```

Comment:
Why do you make a copy every time here for exs array? Each such lookup is O(n), did you consider using a map for deduplication purposes?

Could you also name exs differently, it is not clear what it means from the function signature standpoint.

Reply:
sorry. i dont think  becaose this is tiny and quick-hacks